src/Sulu/Bundle/RouteBundle/Generator/SymfonyExpressionTokenProvider.php

```diff
@@ -38,7 +38,12 @@ public function __construct(TranslatorInterface $translator)
     public function provide($entity, $name)
     {
         try {
-            return $this->expressionLanguage->evaluate($name, ['object' => $entity, 'translator' => $this->translator]);
+            $locale = $this->translator->getLocale();
+            $this->translator->setLocale($entity->getLocale());
```

Comment:
We can not be sure that `$entity->getLocale()` exist @wachterjohannes what do you think should we go with:

```
if (method_exists($entity, 'getLocale')) {

}
```

Reply:
@alexander-schranz your right - in a "normal" usecase with routebundle it will have a `getLocale` but yes the method allows also other entities

@adriansieber will you add this condition?

Reply:
Apart from that, don't you think setting the locale this way might lead to strange bugs? When somebody catches the `CannotEvaluateTokenException` below, the wrong state is kept. I don't know how this is used afterwards, but the `trans` function also accepts a `locale` attribute. Maybe we can decorate this service somehow, so that this `$locale` attribute is always passed?

Reply:
@danrot think we should reset the translator locale in finally.

Reply:
Yeah, at least we should that. Would still prefer not changing it in the first place :see_no_evil: